relay-server/src/services/processor.rs

```diff
@@ -1274,12 +1274,22 @@ impl EnvelopeProcessorService {
         event::finalize(state, &self.inner.config)?;
         self.light_normalize_event(state)?;
         dynamic_sampling::normalize(state);
-        event::filter(state)?;
+        let global_config = self.inner.global_config.current();
+        event::filter(state, global_config.filters())?;
         dynamic_sampling::run(state, &self.inner.config);
 
+        // Don't extract metrics if relay can't apply generic inbound filters.
+        // An inbound filter applied in another up-to-date relay in chain may
+        // need to drop the event, and there should not be metrics from dropped
+        // events.
+        let supported_generic_filters = state.project_state.config.filter_settings.generic.version
```

Comment:
This logic should be hidden away in the `relay-filters` module and `GENERIC_FILTERS_VERSION` shouldn't be public.

And maybe a crazy idea: what if `event::filter` returns an additional struct/enum containing that info?

Reply:
Generally, replacing booleans with enums in function arguments increases readability. In this case, it's a supported/unsupported decision as the function return type, so I think a boolean is a better choice. What do you think?

Reply:
I like the enum 👍

Reply:
I've looked into it, and still think the use-case is simple enough to deal with a boolean. Let me know if you're strongly opinionated on this and feel free to block the PR in that case.